packages/cardpay-sdk/sdk/scheduled-payment-module.ts

```diff
@@ -135,4 +136,97 @@ export default class ScheduledPaymentModule {
 
     return { txs: [transaction, setGuardTransaction], expectedModuleAddress };
   }
+
+  async generateSchedulePaymentSignature(
+    senderSafeAddress: string,
+    spSafeModuleAddress: string,
+    tokenAddress: string,
+    spHash: string
+  ) {
+    let contract = new this.layer2Web3.eth.Contract(ScheduledPaymentABI as AbiItem[], spSafeModuleAddress);
+    let payload = await contract.methods.schedulePayment(spHash).encodeABI();
+
+    let estimate = await gasEstimate(
+      this.layer2Web3,
+      senderSafeAddress,
+      spSafeModuleAddress,
+      '0',
+      payload,
+      Operation.CALL,
+      tokenAddress
+    );
+
+    let from = (await this.layer2Web3.eth.getAccounts())[0];
+    let nonce = getNextNonceFromEstimate(estimate);
+
+    let signatures = await signSafeTx(
+      this.layer2Web3,
+      senderSafeAddress,
+      spSafeModuleAddress,
+      payload,
+      Operation.CALL,
+      estimate,
+      nonce,
+      from,
+      this.layer2Signer
+    );
+
+    return [signatures, nonce, estimate, payload];
+  }
+
+  async schedulePayment(
```

Comment:
Actually now that I look at this I realized I forgot to add the optional `signature` param here (look at my other comment for context). Will add it.